lib/dry/configurable/class_methods.rb

```diff
@@ -47,7 +55,7 @@ def setting(*args, **options, &block)
       #
       # @api public
       def settings
-        @settings ||= Set[*_settings.map(&:name)]
+        Set[*_settings.map(&:name)]
```

Comment:
We don't want to memoize this because it'll capture stale data if called before all settings are defined.

Reply:
@timriley I would remove this, we have `_settings` and this one should be exposed as `settings` IMO. This Set is mostly here for historical reasons, it was just always like that but I'm not sure if this is a valuable thing to keep.

Reply:
@timriley the only reason why I introduced `_settings` was because I didn't want to break backward compatibility by changing `settings` from a set of symbol names to an array of setting objects.